src/native/libs/System.Security.Cryptography.Native/pal_ecc_import_export.c

```diff
@@ -592,3 +601,233 @@ EC_KEY* CryptoNative_EcKeyCreateByExplicitParameters(
+
+    EC_POINT* G = NULL;
+    BIGNUM* xBn = BN_new();
+    BIGNUM* yBn = BN_new();
+    BIGNUM* pBn = NULL;
+    BIGNUM* aBn = NULL;
+    BIGNUM* bBn = NULL;
+    BIGNUM* orderBn = NULL;
+    BIGNUM* cofactorBn = NULL;
+    size_t sufficientSeedBufferSize = 0;
+    unsigned char* seedBuffer = NULL;
+    EC_GROUP* group = NULL;
+    size_t generatorBufferSize = 0;
+    unsigned char* generatorBuffer = NULL;
+
+    // Exit if CryptoNative_EvpPKeyGetEcKeyParameters failed
+    if (rc != 1)
+        goto error;
+
+    if (!xBn || !yBn)
+        goto error;
+
+    int curveTypeNID = EvpPKeyGetCurveType(pkey);
+    if (!curveTypeNID)
+        goto error;
+
+    // Extract p, a, b
+    if (!EVP_PKEY_get_bn_param(pkey, OSSL_PKEY_PARAM_EC_P, &pBn))
+        goto error;
+
+    if (!EVP_PKEY_get_bn_param(pkey, OSSL_PKEY_PARAM_EC_A, &aBn))
+        goto error;
+
+    if (!EVP_PKEY_get_bn_param(pkey, OSSL_PKEY_PARAM_EC_B, &bBn))
+        goto error;
+
+    // curveTypeNID will be always NID_X9_62_characteristic_two_field or NID_X9_62_prime_field
+    group = EC_GROUP_new_by_curve_name(curveTypeNID);
+
+    // In some cases EVP_PKEY_get_field_type can return NID_undef
+    // and some providers seem to be ignoring OSSL_PKEY_PARAM_EC_FIELD_TYPE.
+    // This is specifically true for tpm2 provider.
+    // We can reliably get the field type from the EC_GROUP.
+    int fieldTypeNID = EC_GROUP_get_field_type(group);
+
+    *curveType = NIDToCurveType(fieldTypeNID);
+    if (*curveType == Unspecified)
+        goto error;
+
+    if (!group)
+        goto error;
+
+    if (!EVP_PKEY_get_octet_string_param(pkey, OSSL_PKEY_PARAM_EC_GENERATOR, NULL, 0, &generatorBufferSize))
+        goto error;
+
+    generatorBuffer = (unsigned char*)OPENSSL_malloc(generatorBufferSize);
+    if (!EVP_PKEY_get_octet_string_param(pkey, OSSL_PKEY_PARAM_EC_GENERATOR, generatorBuffer, generatorBufferSize, &generatorBufferSize))
+        goto error;
+
+    G = EC_POINT_new(group);
+
+    if (!G)
+        goto error;
+
+    if (!EC_POINT_oct2point(group, G, generatorBuffer, generatorBufferSize, NULL))
+        goto error;
+
+    if (!EcPointGetAffineCoordinates(group, *curveType, G, xBn, yBn))
+        goto error;
+
+    // Extract order (n)
+    if (!EVP_PKEY_get_bn_param(pkey, OSSL_PKEY_PARAM_EC_ORDER, &orderBn))
+        goto error;
+
+    // Extract cofactor (h)
+    if (!EVP_PKEY_get_bn_param(pkey, OSSL_PKEY_PARAM_EC_COFACTOR, &cofactorBn))
+        goto error;
+
+    // Extract seed (optional)
+    if (!seed || !EVP_PKEY_get_octet_string_param(pkey, OSSL_PKEY_PARAM_EC_SEED, NULL, 0, &sufficientSeedBufferSize))
+    {
+        *seed = NULL;
+        *cbSeed = 0;
+    }
+    else
+    {
+        seedBuffer = (unsigned char*)OPENSSL_malloc(sufficientSeedBufferSize);
+        size_t actualSeedSize;
+        if (EVP_PKEY_get_octet_string_param(pkey, OSSL_PKEY_PARAM_EC_SEED, seedBuffer, sufficientSeedBufferSize, &actualSeedSize))
+        {
+            *seed = seedBuffer;
```

Comment:
To better align with what the EC_KEY export is doing, just wrap the seed in a BN.